frontend/src/utilities/__tests__/useDraggableTable.spec.ts

```diff
@@ -0,0 +1,322 @@
+
+    const { tableProps } = renderResult.result.current;
+    const { onDragOver, ref } = tableProps.tbodyProps;
+
+    ref.current = document.body.querySelector('tbody');
+
+    act(() => {
+      onDragOver({
+        preventDefault: jest.fn(),
+        clientX: 50,
+        clientY: 50,
+        target: inValidElement,
+      } as unknown as React.DragEvent<HTMLTableSectionElement>);
+    });
+
+    expect(tableProps.tbodyProps.ref).toBe(ref);
+  });
+});
+describe('onDragEnd', () => {
+  it('should handle drag end behaviour correctly', () => {
+    const itemOrder = ['item3', 'item1', 'item2'];
+    const tableElement = generateRowsWithItems(itemOrder);
+    document.body.appendChild(tableElement);
+
+    const renderResult = testHook(useDraggableTable)(itemOrder, setItemOrder);
+
+    const { rowProps, tableProps } = renderResult.result.current;
+    const { onDragEnd } = rowProps;
+    const { ref } = tableProps.tbodyProps;
+    ref.current = document.body.querySelector('tbody');
+
+    expect(renderResult.result.current.tableProps.tbodyProps.ref.current).toBe(ref.current);
+
+    const dragEndEvent = {
+      target: {
+        id: 'item3',
+        classList: { remove: jest.fn() },
+        setAttribute: jest.fn(),
+      },
+    } as unknown as React.DragEvent<HTMLTableRowElement>;
+
+    act(() => {
+      onDragEnd(dragEndEvent);
+    });
+    expect(renderResult).hookToHaveUpdateCount(1);
+    expect((dragEndEvent.target as HTMLTableRowElement).classList.remove).toHaveBeenCalled();
+    expect((dragEndEvent.target as HTMLTableRowElement).setAttribute).toHaveBeenCalledWith(
+      'aria-pressed',
+      'false',
+    );
+  });
+});
+describe('onDragLeave', () => {
+  const dragLeaveEvent = {} as unknown as React.DragEvent<HTMLTableSectionElement>;
+
+  it('should handle drag leave behaviour correctly with invalid event', () => {
+    const itemOrder = ['item1', 'item2', 'item3'];
+    const renderResult = testHook(useDraggableTable)(itemOrder, setItemOrder);
+
+    const { tableProps } = renderResult.result.current;
+    const { onDragLeave, ref } = tableProps.tbodyProps;
+    ref.current = null;
+
+    act(() => {
+      onDragLeave(dragLeaveEvent);
+    });
+    expect(renderResult.result.current.tableProps.tbodyProps.ref.current).toBe(null);
+    expect(renderResult.result.current.tableProps.className).toBe(undefined);
+    expect(renderResult).hookToHaveUpdateCount(1);
+  });
+
+  it('should remove the last child if ulNode has children', () => {
+    const itemOrder = ['item4'];
+    const tableElement = generateRowsWithItems(itemOrder);
+    document.body.appendChild(tableElement);
+
+    const renderResult = testHook(useDraggableTable)(itemOrder, setItemOrder);
+    const { tableProps } = renderResult.result.current;
+    const { onDragLeave, ref } = tableProps.tbodyProps;
+
+    ref.current = document.body.querySelector('tbody');
+    expect(renderResult.result.current.tableProps.tbodyProps.ref.current).toBe(ref.current);
```

Comment:
Unnecessary `expect`.
This assertion tests that the assignment on the line above was made....